src/fmnoise/coldbrew.clj

```diff
@@ -43,11 +58,47 @@
                                   weak-values
                                   (.weakValues)
                                   soft-values
-                                  (.softValues))]
+                                  (.softValues)
+                                  eviction-listener
+                                  (.evictionListener
+                                   (reify RemovalListener
+                                     (onRemoval [this key value cause]
+                                       (eviction-listener key value cause))))
+                                  removal-listener
+                                  (.removalListener
+                                   (reify RemovalListener
+                                     (onRemoval [this key value cause]
+                                       (removal-listener key value cause)))))]
     (if cache-fn
       (.build cache-builder (cache-loader cache-fn))
       (.build cache-builder))))
 
+(defn put
+  "Insert/update a cache entry."
+  [^Cache cache key val]
+  (.put cache key val))
+
+(defn put-all
+  "Insert/update multiple cache entries.
+  `m` is a map of key -> value."
+  [^Cache cache m]
+  (.putAll cache m))
+
+(defn invalidate
+  "Invalidate a cache entry."
+  [^Cache cache key]
+  (.invalidate cache key))
+
+(defn get
```

Comment:
seems like we have functionality duplication with `lookup`, so I'm thinking to replace `lookup` implementation with your code, but not sure if `.getIfPresent` works the same as `.get` for LoadingCache 🤔

Reply:
Yeah, it's different. My main problem was that there was no simple `get` for non-loading caches i.e. a get that returns nil if not present. We *could* merge this behaviour with the existing lookup function but then LoadingCache in turn would lack that function again.

Reply:
yeah, then maybe we could end up having something like 
```clj
(if (instance? LoadingCache cache) 
  (.get cache key) 
  (.getIfPresent cache key))
```

Reply:
> caffeine might do the computation in a more efficient/thread safe way than your other lookup arity

yeah, your implementation is better in that sense

Reply:
what do you think about this @JohnnyJayJay ?
```clj
(defn lookup
  "Performs cache lookup. Accepts optional 0-arity function to calculate missing value"
  ([^Cache cache key]
   (if (instance? LoadingCache cache)
     (.get cache key)
     (.getIfPresent cache key)))
  ([^Cache cache key f]
   (.get cache key (reify java.util.function.Function (apply [_ key] (f key))))))

(defn lookup-some
  "Performs cache lookup for existing value. Doesn't trigger missing value recalculation for `LoadingCache`."
  [^Cache cache key] (.getIfPresent cache key))
```

Reply:
if you're ok with this, we can merge your PR and then I'll push this changes, remove `get` and release this to Clojars today

Reply:
> what do you think about this @JohnnyJayJay ?
> ```clj
> (defn lookup
>   "Performs cache lookup. Accepts optional 0-arity function to calculate missing value"
>   ([^Cache cache key]
>    (if (instance? LoadingCache cache)
>      (.get cache key)
>      (.getIfPresent cache key)))
>   ([^Cache cache key f]
>    (.get cache key (reify java.util.function.Function (apply [_ key] (f key))))))
> 
> (defn lookup-some
>   "Performs cache lookup for existing value. Doesn't trigger missing value recalculation for `LoadingCache`."
>   [^Cache cache key] (.getIfPresent cache key))
> ```

The 3-arity lookup change is breaking. Maybe there should be a check whether the provided function takes no arguments, and if so, use the old strategy.

Other than that, looks good.

Reply:
> The 3-arity lookup change is breaking

ah, yeah, it expected 0-arity function. Would be good if we could have easy way to reflect on function  arity https://stackoverflow.com/questions/1696693/clojure-how-to-find-out-the-arity-of-function-at-runtime
rather than something like
```clj
(reify java.util.function.Function
  (apply [_ key]
    (try
      (f key)
      (catch clojure.lang.ArityException _
        (f)))))

Reply:
ok, let's do breaking change then, 1-arity (function of passed key) sounds more reasonable and practical